cmd/bucket-metadata.go

```diff
@@ -46,7 +46,9 @@ const (
 )
 
 var (
-	defaultBucketObjectLockConfig = []byte(`<ObjectLockConfiguration xmlns="http://s3.amazonaws.com/doc/2006-03-01/"><ObjectLockEnabled>Enabled</ObjectLockEnabled></ObjectLockConfiguration>`)
+	enabledBucketObjectLockConfig   = []byte(`<ObjectLockConfiguration xmlns="http://s3.amazonaws.com/doc/2006-03-01/"><ObjectLockEnabled>Enabled</ObjectLockEnabled></ObjectLockConfiguration>`)
```

Comment:
In general I am not a big fan of including default values. 

They are taking up space in memory (and a bit on disk) for no real reason, whereas if `len(field) == 0 {/*default*/}` is easy to check AND doesn't require an unmarshal to get the actual values.

It also makes it clear whether it is a user applied default or a system default, which can be relevant if defaults should change.

Reply:
There is nothing we can do about it - S3 APIs some have default values already on the bucket. So this sort of designates that.

For example versioning config is already present just that the value is disabled by default.

Reply:
I can move it and keep it like allocating structs individually, I can do that too. Since this is minor memory usage I don't know if we should really worry about it.

Reply:
I know it needs to return it when requested, but unless the user has set it I don't see a big reason to actually store a value until they actually set something. 

I would expect it still would be possible to return a default config if none is set. I may misunderstand a state, but it seems that we add config if nothing has been set.

Reply:
Yeah with this I wanted to indicate what is the default state of all bucket metadata, rather than adding it in the handler layer.  Since it's small minor memory i thought this is the right place for it..

Reply:
I can change either way, just thought this is more readable.